react-app/src/api/cosmosAPI.ts

```diff
@@ -65,6 +69,42 @@ export const useCosmosAPI = (): ICosmosAPI => {
     };
   }, [chainInfo.currency.coinMinimalDenom, wallet]);
 
+  const getAddressBalance = useCallback(
+    async (address: string) => {
+      const balance = await stargateQuery.getBalance(
+        address,
+        chainInfo.currency.coinMinimalDenom
+      );
```

Comment:
Yes, but ExtendedQuery doesn't have `getBalanceStaked()`.

Reply:
i think most of the functions from StargateClient/StargateSigningClient are aliases/wrappers to existing function in ExtendedQuery, i reckon we should have a single client for the queries. If you foresee it would take a long time i think we can open an issue so we can remove `getBalance` and `getStakedBalance` from cosmosAPI and use ExtendedQuery altogether

Reply:
Hi sorry, I am a little bit confused as to what needs to be done

> i think most of the functions from StargateClient/StargateSigningClient are aliases/wrappers to existing function in ExtendedQuery, i reckon we should have a single client for the queries.

I suppose you are implying we should use ExtendedQuery, in that case it doesnt have the `getStakedBalance` method, do we create our own implementation using ExtendedQuery? 

> If you foresee it would take a long time i think we can open an issue so we can remove getBalance and getStakedBalance from cosmosAPI and use ExtendedQuery altogether

Do you mean creating a new API hook `useExtendedQueryAPI` that returns `getBalance` and `getStakedBalance`?

Reply:
https://github.com/cosmos/cosmjs/blob/c9fad49acaeae82dcc0b5c5ad5b54636c077af65/packages/stargate/src/stargateclient.ts#L265-L301

we can have `getBalance` in `bankAPI` that calls `query.bank.balance(address, searchDenom)` and 
`getStakedBalance` in `stakingAPI` that calls `query.staking.delegatorDelegations(address, startAtKey)` similar to how you did the `getUnstakingAmount` function, where `query` is just `const { query } = useQueryClient();`

Reply:
I see, yes I think I'd prefer to open an issue for this.